test/test-network/systemd-networkd-tests.py

```diff
@@ -0,0 +1,856 @@
+        with open(os.path.join(os.path.join(os.path.join('/sys/class/net/', link), dev), attribute)) as f:
+            return f.readline().strip()
+
+    def link_exits(self, link):
+        return os.path.exists(os.path.join('/sys/class/net', link))
+
+    def link_remove(self, links):
+        for link in links:
+            if os.path.exists(os.path.join('/sys/class/net', link)):
+                subprocess.call(['ip', 'link', 'del', 'dev', link])
+
+    def read_ipv6_sysctl_attr(self, link, attribute):
+        with open(os.path.join(os.path.join(network_sysctl_ipv6_path, link), attribute)) as f:
+            return f.readline().strip()
+
+    def read_ipv4_sysctl_attr(self, link, attribute):
+        with open(os.path.join(os.path.join(network_sysctl_ipv4_path, link), attribute)) as f:
+            return f.readline().strip()
+
+    def copy_unit_to_networkd_unit_path(self, *units):
+        for unit in units:
+            shutil.copy(os.path.join(networkd_ci_path, unit), network_unit_file_path)
+
+    def remove_unit_from_networkd_path(self, units):
+        for unit in units:
+            if (os.path.exists(os.path.join(network_unit_file_path, unit))):
+                os.remove(os.path.join(network_unit_file_path, unit))
+
+    def start_dnsmasq(self):
+        subprocess.check_call('dnsmasq -8 /var/run/networkd-ci/test-dnsmasq-log-file --log-queries=extra --log-dhcp --pid-file=/var/run/networkd-ci/test-test-dnsmasq.pid --conf-file=/dev/null --interface=veth-peer --enable-ra --dhcp-range=2600::10,2600::20 --dhcp-range=192.168.5.10,192.168.5.200 -R --dhcp-leasefile=/var/run/networkd-ci/lease --dhcp-option=26,1492 --dhcp-option=option:router,192.168.5.1 --dhcp-option=33,192.168.5.4,192.168.5.5', shell=True)
+
+        time.sleep(10)
+
+    def stop_dnsmasq(self, pid_file):
+        if os.path.exists(pid_file):
+            with open(pid_file, 'r') as f:
+                pid = f.read().rstrip(' \t\r\n\0')
+                os.kill(int(pid), signal.SIGTERM)
+
+            os.remove(pid_file)
+
+    def search_words_in_file(self, word):
+        if os.path.exists(dnsmasq_log_file):
+            with open (dnsmasq_log_file) as in_file:
+                contents = in_file.read()
+                print(contents)
+                for part in contents.split():
+                    if word in part:
+                        in_file.close()
+                        print("%s, %s" % (word, part))
+                        return True
+        return False
+
+    def remove_lease_file(self):
+        if os.path.exists(os.path.join(networkd_ci_path, 'lease')):
+            os.remove(os.path.join(networkd_ci_path, 'lease'))
+
+    def remove_log_file(self):
+        if os.path.exists(dnsmasq_log_file):
+            os.remove(dnsmasq_log_file)
+
+    def start_networkd(self):
+        subprocess.check_call('systemctl stop systemd-networkd', shell=True)
+        time.sleep(1)
+        subprocess.check_call('systemctl start systemd-networkd', shell=True)
```

Comment:
Why don't you use `systemctl restart systemd-networkd.service`? I got so many messages like following:
```
Warning: Stopping systemd-networkd.service, but it can still be activated by:
  systemd-networkd.socket
```

Reply:
his response was: https://github.com/systemd/systemd/pull/10147#discussion_r219779258

Reply:
Oh, sorry, I have not read all the previous discussions.
But, just adding sleep() after or before restarting seems enough, no?